src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -28,15 +28,37 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.server")
+	log               = golog.LoggerFor("flashlight.server")
+	frontingProviders = map[string]func(*http.Request) bool{
+		// WARNING: If you add a provider here, keep in mind that Go's http
+		// library normalizes all header names so the first letter of every
+		// dash-separated "word" is uppercase while all others are lowercase.
+		// Also, try and check more than one header to lean on the safe side.
+		"cloudflare": func(req *http.Request) bool {
+			return hasHeader(req, "Cf-Connecting-Ip") || hasHeader(req, "Cf-Ipcountry") || hasHeader(req, "Cf-Ray") || hasHeader(req, "Cf-Visitor")
+		},
+		"cloudfront": func(req *http.Request) bool {
+			return hasHeader(req, "X-Amz-Cf-Id") || headerMatches(req, "User-Agent", "Amazon Cloudfront")
+		},
+	}
```

Comment:
In this case I don't think we should sacrifice current functionality for fear of a hypothetical problem in the future.

- Most Lantern clients will be old for a while.  If we don't support old clients, blocking our current front would leave out the vast majority of our users.  I think if that was actually OK then adding new fronts wouldn't be a priority right now.

- I don't suppose we'll be adding any new fronts (except perhaps Azure, but it makes little sense to multifront that) before we have auto-updates.  After that, we can transition to the scheme you suggest very quickly if needed.

- In extreme cases we can still use fronts that we can't detect.  If we know a front is the only unblocked one we can remove all others from the `frontfqdns` and it will always be selected.